examples/fargate-serverless/main.tf

```diff
@@ -268,27 +265,18 @@ module "vpc" {
 }
 
 #---------------------------------------------------------------
-# Optional - Sample App
+# Sample App
 #---------------------------------------------------------------
-data "kubectl_path_documents" "sample_app" {
-  pattern = "${path.module}/sample-app/*.yaml"
-}
 
 resource "kubernetes_namespace_v1" "sample_app" {
-  count = var.deploy_sample_app ? 1 : 0
   metadata {
-    name = local.smaple_app_namespace
+    name = local.sample_app_namespace
   }
-  depends_on = [
-    module.eks_blueprints,
-    module.eks_blueprints_kubernetes_addons
-  ]
 }
 
 resource "kubectl_manifest" "sample_app" {
-  count     = var.deploy_sample_app ? length(data.kubectl_path_documents.sample_app.documents) : 0
-  yaml_body = element(data.kubectl_path_documents.sample_app.documents, count.index)
-  depends_on = [
-    kubernetes_namespace_v1.sample_app[0]
-  ]
+  yaml_body = templatefile("sample_app.yaml", {
```

Comment:
flattened to one multi-resource yaml file with templating so that we can map dependencies (by passing namespace from namespace resource)